File: src/main/scala/dev/vrba/harold/modules/sample/SampleCommand.scala
package dev.vrba.harold.modules.sample

import dev.vrba.harold.core.commands.{Command, CommandResult, Success}
import net.dv8tion.jda.api.entities.Message

class SampleCommand extends Command {
  override val name: String = "sample"
  override val aliases: Array[String] = Array("samplealias")

  override def execute(message: Message): CommandResult = {
    message.addReaction("☺").queue()
    Success()
  }
}

File: src/main/kotlin/dev/vrba/harold/modules/sample/commands/SampleCommand.kt
package dev.vrba.harold.modules.sample.commands

import dev.vrba.harold.core.commands.Command
import dev.vrba.harold.core.commands.CommandExecutionResult
import dev.vrba.harold.core.commands.CommandExecutionResult.Success
import dev.vrba.harold.core.commands.CommandScope
import net.dv8tion.jda.api.events.message.MessageReceivedEvent
import org.springframework.stereotype.Component

@Component
class SampleCommand : Command {
    override val name = "sample"

    override val aliases = listOf("samplealias")

    override val scope = CommandScope.GuildsOnly

    override val permissions = emptyList<String>()

    override fun execute(event: MessageReceivedEvent): CommandExecutionResult {
        event.channel.sendMessage("Executed the sample command")
        return Success
    }
}